KsApi/MockService.swift

```diff
@@ -702,19 +695,26 @@
       )
     }
 
-    internal func fetchGraphCategories(query _: NonEmptySet<Query>)
-      -> SignalProducer<RootCategoriesEnvelope, GraphError> {
-      if let error = self.fetchGraphCategoriesError {
-        return SignalProducer(error: error)
-      } else if let response = self.fetchGraphCategoriesResponse {
-        return SignalProducer(value: response)
+    internal func fetchGraphCategories()
+      -> SignalProducer<RootCategoriesEnvelope, ErrorEnvelope> {
+      guard let client = self.apolloClient else {
+        return .empty
       }
-      return SignalProducer(value: RootCategoriesEnvelope.template)
+
+      let fetchGraphCategoriesQuery = GraphAPI.FetchRootCategoriesQuery(withParentCategoryAnalyticsName: true)
+
+      return client.fetchWithResult(query: fetchGraphCategoriesQuery, result: self.fetchGraphCategoriesResult)
```

Comment:
Noticed we're only using this syntax with the requests you've edited in this PR. Do we need to retrofit the other mock service methods with this implementation too? If so we should have a ticket in the backlog so we remember.

Reply:
Yea good call, most of the methods are already using `client.fetchWithResult` already, ones missing are:
`facebookConnect(facebookAccessToken`
`fetchCommentReplies(query` (will be handled in upcoming ticket)
`fetchConfig()` (not sure if this is a network call necessarily, need to investigate)
`fetchFriends()`
`fetchFriendStats()`
`followAllFriends()`
`exportData()`
`followFriend(userId`
`fetchErroredUserBackings(status` (there is a note about this needing it's return envelope to be decodable)
`fetchGraph<A>` (use case)
`unfollowFriend(userId`
`login`
`logout`
`fetchActivities(count`
`fetchBacking(forProject`
`backingUpdate(forProject`
...there's a lot more...and I just realized a lot of these are v1 calls, so I think as we move those to GQL we'll do the refactoring here as well (have to, project will fail to build otherwise).
All of the GQL calls though should be using the `client.fetchWithResult(` method above, and that should be complete once we remove the DSL.